### ISSUE TITLE
takePictureAsync does not return if focus changed

### ISSUE THREAD

**Issue by cristianoccazinsp** [react-native-community/react-native-camera#2421]
# Bug Report

**To Do First**

- [x] Did you try latest release? Yes
- [x] Did you try master? Yes
- [x] Did you look for existing matching issues? Yes

**Platforms**

Android

**Versions**

- Android: 8 and 9
- react-native-camera: 3.1.0
- react-native: 0.60.4
- react: 16.8.6
  

**Description/Current Behaviour**
If I call setState to change ```autoFocusPointOfInterest``` while the camera is trying to focus to capture after a call too ```takePictureAsync```, the method never resolves and neither rejects, it basically becomes stuck.

**Expected Behaviour**
takePictureAsync should always resolve or reject.

**Steps to Reproduce**

The code to capture looks like:
```
 try{
        data = await this.camera.takePictureAsync(options);
      }
catch(err){
        Alert.alert("Error", "Failed to take picture: " + err.message);
}
```

if a setState is called to update the autoFocusPointOfInterest RNCamera property before the above method resolves, it will never resolve. There are no errors on the android console neither.

**Comment by cristianoccazinsp** [comment]
Update: If flash is turned on, flash/light also gets stuck until the camera component is unmounted. Tested on both Pixel 2 and Moto G5. This makes tap to focus nearly impossible to use on Android.

**Comment by cristianoccazinsp** [comment]
Related to this: https://github.com/react-native-community/react-native-camera/issues/2420

If we remove the call to autoFocus before capturing all issues go away.
https://github.com/react-native-community/react-native-camera/blob/master/android/src/main/java/com/google/android/cameraview/Camera1.java#L453

**Comment by cristianoccazinsp** [comment]
In addition to the above, if we don't remove the autoFocus call, this is most likely related to the additional calls to autoFocus and cancel/reset auto focus. The capture event is most likely lost when this happens and the promise is never resolved. Following https://github.com/react-native-community/react-native-camera/issues/2420, we most likely want to just get rid of the auto focus call (perhaps use a setting for it), which would solve both issues.

**Comment by cristianoccazinsp** [comment]
Related changes: https://github.com/react-native-community/react-native-camera/pull/2427

**Comment by n1ru4l** [comment]
Resolved in #2427
